src/fragments/lib/graphqlapi/flutter/subscribe-data.mdx

```diff
@@ -53,10 +53,130 @@ void unsubscribe() {
 }
 ```
 
-Note that in addition to an `onCreate` subscription, you can also call `.onUpdate()` or `.onDelete()`.
+In addition to an `onCreate` subscription, you can also call `.onUpdate()` or `.onDelete()`.
 
 ```dart
 final onUpdateSubscriptionRequest = ModelSubscriptions.onUpdate(Todo.classType);
 // or
 final onDeleteSubscriptionRequest = ModelSubscriptions.onDelete(Todo.classType);
 ```
+
+## Subscription connection status
+
+Now that you set up the application and are using subscriptions, you may want to know when the subscription is closed, or reflect to your users when the subscription isn’t healthy. You can monitor the subscription status for changes via `Amplify.Hub`
+
+```dart
+Amplify.Hub.listen(
+  HubChannel.Api,
+  (ApiHubEvent event) {
+    if (event is SubscriptionHubEvent) {
+      safePrint(event.status);
+    }
+  },
+);
+```
+
+#### SubscriptionStatus
+
+- **`connected`** - Connected and working with no issues
+- **`connecting`** - Attempting to connect (both initial connection and reconnection)
+- **`pendingDisconnect`** - Connection has no active subscriptions and is shutting down
+- **`disconnected`** - Connection has no active subscriptions and is disconnected
+- **`failed`** - Connection had a failure and has been disconnected
+
+## Automated Reconnection
+
+Under the hood, we will attempt to maintain a healthy web socket connection through network changes. For example, if a device’s connection changes from Wi-Fi to 5g network, the plugin will attempt to reconnect using the new network.
+
+Likewise, when disconnected from the internet unexpectedly, the plugin will attempt to reconnect using an exponential retry/back off strategy. By default, we will make 8 recovery attempts over about 50 seconds. If we cannot make a successful connection, then the web socket will be closed. You can customize this strategy when configuring the API plugin through `RetryOptions`.
+
+```dart
+Future<void> _configureAmplify() async {
+  final apiPlugin = AmplifyAPI(
+    modelProvider: ModelProvider.instance,
+    // Optional config
+    subscriptionOptions: const GraphQLSubscriptionOptions(
+      retryOptions: RetryOptions(maxAttempts: 10),
+    ),
+  );
+  await Amplify.addPlugin(apiPlugin);
+
+  try {
+    await Amplify.configure(amplifyconfig);
+  } on AmplifyAlreadyConfiguredException {
+    safePrint(
+        "Tried to reconfigure Amplify; this can occur when your app restarts on Android.");
+  }
+}
+```
+
+While offline, your application will miss messages and will not automatically catch up when reconnection happens. Depending on your use case, you may want to take action to catch up when your app comes back online.
```

Comment:
let us make this a callout please

Reply:
let us make it clear the below code snippet is to show an example of how to handle pulling todos after a reconnect.